data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ConditionalExpressionEvaluatorIT.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ConditionalExpressionEvaluatorIT {
+
+    private AnnotationConfigApplicationContext applicationContext;
+
+    @BeforeEach
+    void beforeEach() {
+        applicationContext = new AnnotationConfigApplicationContext();
+        applicationContext.scan("org.opensearch.dataprepper.expression");
+        applicationContext.refresh();
+    }
+
+    @Test
+    void testConditionalExpressionEvaluatorBeanAvailable() {
+        final ConditionalExpressionEvaluator evaluator = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+        assertThat(evaluator, isA(ConditionalExpressionEvaluator.class));
+    }
+
+    @Test
+    void testConditionalExpressionEvaluatorBeanNotSingleton() {
+        final ConditionalExpressionEvaluator instanceA = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+        final ConditionalExpressionEvaluator instanceB = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+        assertThat(instanceA, not(is(instanceB)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("validExpressionArguments")
+    void testConditionalExpressionEvaluator(final String expression, final Event event, final Boolean expected) {
+        final ConditionalExpressionEvaluator evaluator = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+
+        final Boolean actual = evaluator.evaluate(expression, event);
+
+        assertThat(actual, is(expected));
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidExpressionArguments")
+    void testConditionalExpressionEvaluatorThrows(final String expression, final Event event, final Class<? extends Throwable> throwable) {
+        final ConditionalExpressionEvaluator evaluator = applicationContext.getBean(ConditionalExpressionEvaluator.class);
+
+        assertThrows(throwable, () -> evaluator.evaluate(expression, event));
+    }
+
+    private static Stream<Arguments> validExpressionArguments() {
+        return Stream.of(
+                Arguments.of("true", event("{}"), true),
+                Arguments.of("/status_code == 200", event("{\"status_code\": 200}"), true),
```

Comment:
It might be a good idea to test that `/status_code == 200` is false when the `status_code` is not present (ie. null).